File: 3sem/C++ i STL/lista7/zad2.cpp
#include <iostream>
#include <random>
#include <chrono>
#include <fstream>
#include <map>
#include <cmath>
using namespace std;

static default_random_engine dre;

void uniform_dsb() {
    // equally probable random numbers in [0.0, 1.0)
    uniform_real_distribution<double> distribution(0.0, 1.0);
    map <double, int> count;
    for (int i = 0; i < 1000; i ++)
        count[distribution(dre)] ++;
    ofstream file;
    file.open("unfiorm.csv");
    for (auto x : count)
        file << x.first << "," << x.second << endl;
    file.close();

}

void binomal_dsb() {
    // binomial distribution in [0, 9]
    binomial_distribution<int> distribution(9, 0.5);
    map <int, int> count;
    for (int i = 0; i < 1000; i ++)
        count[distribution(dre)] ++;
    ofstream file;
    file.open("binomial.csv");
    for (auto x : count)
        file << x.first << "," << x.second << endl;
    file.close();
}

void normal_dsb() {
    //normal distribution with mean 5.0, standard deviation 2.0
    normal_distribution<double> distribution{0.0, 1.0};
    map <int, int> count; // numbers of elements in [floor(x), ceiling(x) )
    for (int i = 0; i < 1000; i ++)
    {
        int tmp = floor(distribution(dre));
        //tmp is actually sumbol of range [floor(x), ceiling(x)]
        count[tmp] ++;
    }
    ofstream file;
    file.open("normal.csv");
    for (auto x : count)
        file << x.first << "," << x.second << endl;
    file.close(); 
}  

int main() {

    dre.seed(chrono::system_clock::now().time_since_epoch().count());
    uniform_dsb();
    binomal_dsb();
    normal_dsb();
    return 0;
}

File: 4sem/RPIS/lista3/zad2.py
days = {0 : 0,1 : 0,2 : 0,3 : 0,4 : 0,5 : 0,6 : 0}

normal =   [0,3,3,6,1,4,6,2,5,0,3,5]
not_normal=[0,3,4,0,2,5,0,3,6,1,4,6]

act = 6

for j in range (1601,2001):
	if j % 4 == 0 and (j % 100 != 0 or j % 400 == 0):
		for i in not_normal:
			days[(act + i) % 7] += 1
		act = (act + 6 + 3) % 7
	else:
		for i in normal:
			days[(act + i) % 7] += 1
		act = (act + 5 + 3) % 7

print(days)


# if j % 4 == 0 and (j % 100 != 0 or j % 400 == 0):
# if j % 4 != 0 or (j % 100 == 0 and j % 400 != 0):
